### DISCUSSION TITLE
Nariman's idea: Public Test Suites

### DISCUSSION THREAD

**Discussion by orchestr7** [saveourtool/save-cloud#916] [Ideas]
@petertrr @sanyavertolet @icemachined - @nulls  has two ideas:
1) To move entity test suites to the Organization entity: he wants to cover cases when ktlint and diktat are using same suites uploaded to organization. I do not like this idea, because it looks more convenient to have tests on the level of the project.
2) Second idea is more interesting: he proposes OPEN PUBLIC test suites. These test suites could be shared by organization for other organizations (as well as predefined STANDARD test suites)

What do you think?

**Comment by sanyavertolet** [comment]
Not sure about *moving* test suites to organization level, but it might be good to support organization test suites.

Shareable suites seem to be a good idea. I guess the form for uploading contest suites might be useful here.

All mentioned above make me think of a component like `ManageRolePermissionCard` to manage testSuites as well as users in groups and in global scope

**Comment by nulls** [comment]
> To move entity test suites to the Organization entity

For me, TestSuites is different type of entity. Also, there are organization which provides only TestSuites for testing (like contest) without developing own tool

**Comment by nulls** [comment]
Agreed to save test suites on Organization level, but will add managing access for projects by roles in phase 2
